### PR TITLE
Remove build hack

### PR DESCRIPTION
**Pull request by sbrunner**
Base on  #2888

The hack is adding `> /tmp/logs 2>&1 && touch success) || true)` in the build step and failed later...

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,19 +2,61 @@ GIT_HEAD_ARG = --build-arg=GIT_HEAD=$(shell git rev-parse HEAD)
...
+build-builder:
+	docker build $(GIT_HEAD_ARG) --target=builder --tag=mapfish_print_builder .
+
+.PHONY: checks
+checks: spotbugs violations
+
+# --volume=$(PWD):/src mapfish_print_builder
+.PHONY: spotbugs
+spotbugs: build-builder
+	docker run --rm --detach \
```
Comment (sbrunner) [inline Makefile:23]: don't use --detach

File: Makefile [hunk 2]
```diff
@@ -2,19 +2,61 @@ GIT_HEAD_ARG = --build-arg=GIT_HEAD=$(shell git rev-parse HEAD)
...
+		mapfish_print_builder gradle :core:spotbugsMain
+
+.PHONY: violations
+violations: build-builder
+	docker run --rm \
+		mapfish_print_builder gradle :core:violations
+
+
+.PHONY: profile
+profile: build-builder ## Profile a task
```
Comment (sbrunner) [inline Makefile:33]: The result will be in build/report

File: Makefile [hunk 3]
```diff
@@ -2,19 +2,61 @@ GIT_HEAD_ARG = --build-arg=GIT_HEAD=$(shell git rev-parse HEAD)
...
+	docker stop print-builder || true
+	docker run --rm --detach --name=print-builder mapfish_print_builder tail --follow /dev/null
+
+	# Change this line to run the task you want to profile
+	docker exec print-builder gradle --profile :core:test
+
+	# Get the profile
+	rm -rf build/reports/ || true
+	mkdir -p build/
+	docker cp print-builder:/src/build/reports/ build/
```
Comment (sbrunner) [inline Makefile:43]: missing docker stop print-builder

**Comment by sbrunner** [comment]
We should keep check style...

File: .github/workflows/main.yaml [hunk 4]
```diff
@@ -82,60 +82,27 @@ jobs:
...
           path: examples/build/resources/test/examples
           if-no-files-found: ignore
-        if: failure()
-
-      - name: Collect test results
-        run: |
-          mkdir -p /tmp/test_results/junit
-          find . -name '*TEST-*.xml' -exec cp -v {} /tmp/test_results/junit/ \;
-        if: failure()
-      - uses: actions/upload-artifact@v3
```
Comment (sebr72) [inline .github/workflows/main.yaml:134]: As discussed please reintroduce the 4 CI artifacts

File: Dockerfile [hunk 5]
```diff
@@ -3,44 +3,38 @@ FROM gradle:8.1.1-jdk11 AS builder
...
 ARG GIT_HEAD
 ENV GIT_HEAD=${GIT_HEAD}
 
-COPY publish ./publish
-
+# Exclude the tasks that will run our of the docker build (in a docker run)
 RUN --mount=type=cache,target=/home/gradle/.gradle \
-   ([ -e success ] && ( (gradle :publish:build >> /tmp/logs 2>&1) && touch success-publish)) || true
+   gradle --exclude-task=:core:test --exclude-task=:core:spotbugsMain --exclude-task=:core:checkstyleMain \
+   :core:build :core:explodedWar :publish:build :examples:build buildDocs
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply Dockerfile]: `core:explodedWar` is steel needed...
Others done :-)

File: core/src/main/java/org/mapfish/print/HumanAlphaSerie.java
Comment (sebr72) [inline core/src/main/java/org/mapfish/print/HumanAlphaSerie.java:1]: Let's move this to a sub-package like jasperreports

**Review by sebr72 — CHANGES_REQUESTED** [review]

**Review by sebr72 — APPROVED** [review]
Approved pending requested jasperreports package creation

**Pull request merged**